corehq/apps/users/models.py

```diff
@@ -230,6 +230,11 @@ class UserRole(Document):
     permissions = SchemaProperty(Permissions)
     is_archived = BooleanProperty(default=False)
 
+    @classmethod
+    @quickcache(['doc_id'], timeout=0)
+    def get(cls, doc_id):
```

Comment:
Just offlined with Danny.  I'll come back to this and make a `QuickcachedDocumentMixin` which does this but has the proper invalidation like `CachedCouchDocumentMixin`.